src/lint/rules/spacing.rs

```diff
@@ -1,34 +1,148 @@
 use itertools::izip;
 use std::convert::TryInto;
 use serde::{Deserialize, Serialize};
+use crate::analysis::parsing::lexer::TokenKind;
+use crate::analysis::parsing::misc::CDeclContent;
 use crate::analysis::parsing::types::{BitfieldsContent, LayoutContent,
                                       StructTypeContent};
 use crate::lint::{rules::{Rule, RuleType},
                   DMLStyleError};
-use crate::analysis::parsing::tree::{TreeElement, ZeroRange};
-use crate::analysis::parsing::expression::{FunctionCallContent, IndexContent,
+use crate::analysis::parsing::tree::{LeafToken, TreeElement, ZeroRange};
+use crate::analysis::parsing::expression::{BinaryExpressionContent,
+                                           FunctionCallContent, IndexContent,
                                            PostUnaryExpressionContent,
+                                           TertiaryExpressionContent,
                                            UnaryExpressionContent};
-use crate::analysis::parsing::statement::{CompoundContent,
-                                          ExpressionStmtContent,
-                                          IfContent, VariableDeclContent};
+use crate::analysis::parsing::statement::{AfterContent, CompoundContent, ExpressionStmtContent,
+                                          ForContent, IfContent, VariableDeclContent,
+                                          WhileContent};
 use crate::analysis::parsing::structure::{MethodContent,
                                           ObjectStatementsContent};
 
 use crate::span::{ZeroIndexed, Range};
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct SpReservedOptions {}
+
+pub struct SpReservedRule {
+    pub enabled: bool,
+}
+
+pub struct SpReservedArgs {
+    before_range: Option<ZeroRange>,
+    token_range: ZeroRange,
+    after_range: Option<ZeroRange>,
```

Comment:
Judging from the from_ family of methods below, 'after_range' will always be Some(_). Is there a future case where it would not be? If not, it should not be an Option value.

Reply:
That seems correct. We are testing this locally without the Option and its working as expected. Will push this change.